### ISSUE TITLE
db_url does not handle '#' in password?

### ISSUE THREAD

**Issue by lisac** [joke2k/django-environ#55]
I'm trying to configure a database password that happens to contain a `#`. Is that character not supported in this package's processing of the database url? When I alter the `test_postgres_parsing` in [test.py](https://github.com/joke2k/django-environ/blob/develop/environ/test.py#L237) so that the password contains that character, eg `wegauwhgeuioweg` --> `weg#auwhgeuioweg`, the test encounters this error:

``` ======================================================================
ERROR: test_postgres_parsing (environ.test.DatabaseTestSuite)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\projects\django-environ\environ\test.py", line 241, in test_postgres_parsing
    url = Env.db_url_config(url)
  File "c:\projects\django-environ\environ\environ.py", line 371, in db_url_config
    'PORT': _cast_int(url.port),
  File "C:\Python27\Lib\urlparse.py", line 113, in port
    port = int(port, 10)
ValueError: invalid literal for int() with base 10: 'weg'
```

I've found that other special chars, such as these: `}{)(}|!\^><;` can be embedded in the database password and not break the test. For example, when I alter the password from `wegauwhgeuioweg` to `:wegauw}{)(}|!\^><;hgeuioweg` on lines 238 and 245 of  the test case in [test.py](https://github.com/joke2k/django-environ/blob/develop/environ/test.py#L238), no tests fail.

My environment is python 2.7.10.

Summary: Is it true that a `#` included in the db password will corrupt the db url?

**Comment by joke2k** [comment]
Some characters must be encoded to avoid beaking the url parsing.
See #48

**Comment by joke2k** [comment]
Fixed with #69 
thank you

**Comment by codeasashu** [comment]
This is still issue in 0.4.5

```py

import os
import environ

os.environ['a'] = 'mysql://engineuser:i!kkGhs#$j57@127.0.0.1:3306/mydb'
env.db('a')

url ParseResult(scheme='mysql', netloc='engineuser:i!)(iLisk', path='', params='', query='', fragment='$js7@10.0.0.80:3306/myoperator_test')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/var/app/python/micro_api/user-availability-api/venv/lib/python3.6/site-packages/environ/environ.py", line 204, in db_url
    return self.db_url_config(self.get_value(var, default=default), engine=engine)
  File "/var/app/python/micro_api/user-availability-api/venv/lib/python3.6/site-packages/environ/environ.py", line 403, in db_url_config
    'PORT': _cast_int(url.port) or '',
  File "/usr/lib/python3.6/urllib/parse.py", line 169, in port
    port = int(port, 10)
ValueError: invalid literal for int() with base 10: 'i!kkGhs'
```
